pkg/scheduler/framework/plugins/clusteraffinity/state.go

```diff
@@ -0,0 +1,43 @@
+package clusteraffinity
+
+import (
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework"
+)
+
+type pluginState struct {
+	// requiredAffinityTerms is a list of processed version of required cluster affinity terms.
+	requiredAffinityTerms []framework.AffinityTerm
+
+	// preferredAffinityTerms is a list of processed version of preferred cluster affinity terms.
+	preferredAffinityTerms *framework.PreferredAffinityTerms
+}
+
+// preparePluginState initializes the state for the plugin to use in the scheduling cycle.
+// TODO will call this func in the PreFilter stage.
+func preparePluginState(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (*pluginState, error) {
```

Comment:
For example, this bookkeeping is not necessary as the extension points can read them directly; it does not save any cost.